docs/dev/dev_setup.rst

```diff
@@ -65,6 +65,15 @@ command. You only need to run it once, from within the cloned repo:
 
     $ git config --local merge.ours.driver true
 
+.. note::
+
+   If there have been changes to the environment on a branch (e.g. ``dev``) that you
+   merge into your own branch, the lockfiles will need to be regenerated. This can be
+   done automatically by pushing the merged changes to your branch on GitHub, waiting a
+   couple of minutes for the ``update-conda-lockfile`` GitHub Action to run, and then
+   pulling the fresh lockfiles to your local development environment. You can also
+   regenerate the lockfiles locally (see below).
```

Comment:
Would there be a way to detect this situation and force the regeneration as part of pre-commit checks? I suppose that the current approach works as well, but I find that sometimes the automatic remote changes to my branch can cause some minor git headaches requiring pulls with rebasing.

Reply:
I suspect there are better ways to deal with the lockfile regeneration, but I think we would probably still want something that runs on GitHub to enforce the correspondence between the dependencies in `pyproject.toml` and the lockfiles. In the same way that we use `pre-commit.ci` to ensure that all the pre-commit hooks have actually been run and pass. We've tried to avoid git headaches with the lockfiles in particular by using the "ours" merge driver and you can always just blow them away entirely and recreate them from scratch.